lib/rubocop/cop/style/if_with_boolean_literal_branches.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks for redundant `if` with boolean literal branches.
+      # It checks only conditions to return boolean value (`true` or `false`) for safe detection.
+      # The conditions to be checked are comparison methods, predicate methods, and double negative.
+      #
+      # @example
+      #   # bad
+      #   if foo == bar
+      #     true
+      #   else
+      #     false
+      #   end
+      #
+      #   # bad
+      #   foo == bar ? true : false
+      #
+      #   # good
+      #   foo == bar
+      #
+      class IfWithBooleanLiteralBranches < Base
+        extend AutoCorrector
+
+        MSG = 'Remove redundant %<keyword>s with boolean literal branches.'
+
+        def_node_matcher :if_with_boolean_literal_branches?, <<~PATTERN
+          {
+            (if #return_boolean_value? (true) (false))
+            (if #return_boolean_value? (false) (true))
```

Comment:
You can use `(if #return_boolean_value? {(true) (false) | (false) (true)}) `

Reply:
Cool ❤️